### PR TITLE
Udim support

### PR DESCRIPTION
**Pull request by Vsevolod1983**
TICKET: 1208

PURPOSE:
UDIMS don't supported

EFFECT OF CHANGES:
Now we support "Udim(Mari)" option in FileNode in Maya

### CODE REVIEW

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp [hunk 1]
```diff
@@ -13,27 +13,77 @@ limitations under the License.
...
+{
+	MStringArray fileNames;
+
+	std::ostringstream ostream;
+	ostream << "source \"rprCmdRenderUtils.mel\"; geFileNodeUDIMFiles(\"" << nodeName.asChar() << "\")";
+
+	MStatus status = MGlobal::executeCommand(ostream.str().c_str(), fileNames);
+
+	size_t index = std::string(udimPathPattern.asChar()).find("<UDIM>");
+	size_t tagLength = std::string("UDIM").length();
```
Comment (DMotorygin) [inline FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp:32]: why not a constant?

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp [hunk 2]
```diff
@@ -13,27 +13,77 @@ limitations under the License.
...
+
+	if (index == std::string::npos)
+	{
+		MGlobal::displayWarning(MString("UDIM pattern is not recognized: ") + udimPathPattern);
+		return;
```
Comment (DMotorygin) [inline FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp:37]: maybe move this check to the top of the function before mel command is called?

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp [hunk 3]
```diff
@@ -13,27 +13,77 @@ limitations under the License.
...
+
 frw::Value MayaStandardNodeConverters::FileNodeConverter::Convert() const
 {
 	MPlug texturePlug = m_params.shaderNode.findPlug("computedFileTextureNamePattern");
 	MString texturePath = texturePlug.asString();
 
-	MPlug colorSpacePlug = m_params.shaderNode.findPlug("colorSpace");
+	MPlug uvTilingModePlug = m_params.shaderNode.findPlug("uvTilingMode");
+
+	const int fileNodeUdimMode = 3;
```
Comment (DMotorygin) [inline FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp:59]: please add comment what value 3 stand for

File: FireRender.Maya.Src/frWrap.h [hunk 4]
```diff
@@ -1199,19 +1199,49 @@ namespace frw
...
 			rpr_int res = rprImageSetGamma(Handle(), gamma);
 			checkStatus(res);
 		}
 
+		void SetUDIM(rpr_uint tileIndex, frw::Image image)
+		{
+			rpr_int res = rprImageSetUDIM(Handle(), tileIndex, image.Handle());
+			checkStatus(res);
+
+			data().m_udimsMap[tileIndex] = image;
```
Comment (DMotorygin) [inline FireRender.Maya.Src/frWrap.h:1242]: Maybe add some check that tileIndex is unique?

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp (hunk 3 above)
Comment (Vsevolod1983) [reply FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp:59]: Done

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp (hunk 2 above)
Comment (Vsevolod1983) [reply FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp:30]: Done

File: FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp (hunk 1 above)
Comment (Vsevolod1983) [reply FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp]: Done

File: FireRender.Maya.Src/frWrap.h (hunk 4 above)
Comment (Vsevolod1983) [reply FireRender.Maya.Src/frWrap.h:1242]: No, because technically you can overwrite image by using the same tile index

**Pull request merged**